woocommerce/woocommerce
# Set paths for `allure-results`, `test-results.json`, and save state files to be inside their respective E2E or API folders

### All Submissions:




### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/35239.

Currently, the test outputs of our E2E and API tests are being saved outside of the `plugins/woocommerce/tests` folder. They should probably be stored within their respective E2E and API test directories.

Here is the list of these test outputs, where they're currently being saved in `trunk`, and where this PR would save them in:

| test artifact                                                                             | currently being saved in              | should be saved in                         |
| ----------------------------------------------------------------------------------------- | ------------------------------------- | ------------------------------------------ |
| `allure-results` folder, `test-results.json`, and `storage` folder generated by E2E tests | `plugins/woocommerce/e2e`             | `plugins/woocommerce/tests/e2e-pw`         |
| `allure-results` folder and `test-results.json` generated by API tests                    | `plugins/woocommerce/api-test-report` | `plugins/woocommerce/tests/api-core-tests` |

This PR also untracks the `plugins/woocommerce/tests/e2e/storage` directory and its contents (`adminState.json` and `customerState.json`).

### How to test the changes in this Pull Request:

#### Testing locally

1. Checkout this branch on your local machine.
2. Launch the local e2e wpenv environment.
3. Run this short `basic.spec.js` E2E spec (We don't need to run the entire test suite. We're only concerned where Playwright will save the test outputs.)
   ```bash
   cd plugins/woocommerce
   USE_WPENV=1 pnpm exec playwright test --config=tests/e2e-pw/playwright.config.js basic.spec.js
   ```
4. Wait for the test to finish.
5. Verify that the `allure-results` folder, `test-results.json` file, and `storage` folder are saved inside `plugins/woocommerce/tests/e2e-pw` like so:
   <img width="350" alt="qEqBYyyIU9" src="https://user-images.githubusercontent.com/4509348/197124780-8a5ab1b4-537b-477e-a676-410f84ad2ed7.png">
6. Verify that there's no `plugins/woocommerce/e2e` folder created.
7. Now, run this short `hello.test.js` API test. Again, we don't need to run the entire API test suite, since we're only concerned where the test artifacts will be saved.
   ```bash
   cd plugins/woocommerce
   pnpm playwright test --config=tests/api-core-tests/playwright.config.js ./tests/api-core-tests/tests/hello/hello.test.js
   ```
8. Wait for the test to finish.
9. Verify that the `allure-results` folder and `test-results.json` file are saved inside `plugins/woocommerce/tests/api-core-tests/api-test-results` like so:
   <img width="319" alt="VcDVSISo7R" src="https://user-images.githubusercontent.com/4509348/197124869-cfa649c9-f46e-4c56-84c9-2f34e7748614.png">
10. Verify that there's no `plugins/woocommerce/api-test-results` folder created.

#### Testing in CI

1. Just make sure that the following checks pass:
   - Run tests against PR `(pr-build-and-e2e-tests.yml)`
   - Run tests against PR in an environment with COT enabled `(cot-pr-build-and-e2e-tests.yml)` - this will have to be triggered manually
   - Run daily tests in an environment with COT enabled `(cot-build-and-e2e-tests-daily.yml)` - this will also have to be triggered manually
1. Also verify that test report artifacts were generated, and the _"Test Results Summary"_ were generated with correct stats in each of the workflows mentioned above. This means that our workflows were correctly able to find their new save location.
   <img width="674" alt="NtiLyuIrAf" src="https://user-images.githubusercontent.com/4509348/197124910-15395422-b181-4a5c-b893-627044f3bf69.png">

### Other information:




### FOR PR REVIEWER ONLY:
